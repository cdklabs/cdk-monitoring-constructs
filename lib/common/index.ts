export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
