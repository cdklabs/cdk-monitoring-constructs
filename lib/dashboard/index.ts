export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
