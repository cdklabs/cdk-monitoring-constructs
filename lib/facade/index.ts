export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
