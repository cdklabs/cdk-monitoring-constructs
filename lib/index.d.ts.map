{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}