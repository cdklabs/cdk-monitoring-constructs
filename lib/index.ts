export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
