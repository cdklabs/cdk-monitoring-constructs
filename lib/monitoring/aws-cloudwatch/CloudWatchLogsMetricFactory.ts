import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";

import {
  BaseMetricFactory,
  BaseMetricFactoryProps,
  MetricFactory,
  MetricStatistic,
} from "../../common";

const CloudWatchLogsNamespace = "AWS/Logs";

export interface CloudWatchLogsMetricFactoryProps
  extends BaseMetricFactoryProps {
  /**
   * Name of the log group to monitor.
   */
  readonly logGroupName: string;
}

export class CloudWatchLogsMetricFactory extends BaseMetricFactory<CloudWatchLogsMetricFactoryProps> {
  private readonly dimensionsMap: DimensionsMap;

  constructor(
    metricFactory: MetricFactory,
    props: CloudWatchLogsMetricFactoryProps,
  ) {
    super(metricFactory, props);

    this.dimensionsMap = {
      LogGroupName: props.logGroupName,
    };
  }

  metricIncomingLogEvents() {
    return this.metricFactory.createMetric(
      "IncomingLogEvents",
      MetricStatistic.N,
      "Logs",
      this.dimensionsMap,
      undefined,
      CloudWatchLogsNamespace,
      undefined,
      this.region,
      this.account,
    );
  }
}
