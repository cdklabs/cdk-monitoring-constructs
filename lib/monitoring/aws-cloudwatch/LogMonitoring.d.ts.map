{"version":3,"file":"LogMonitoring.d.ts","sourceRoot":"","sources":["LogMonitoring.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAkB,MAAM,4BAA4B,CAAC;AAErE,OAAO,EACL,mBAAmB,EAGnB,UAAU,EACV,eAAe,EAChB,MAAM,cAAc,CAAC;AAKtB,MAAM,WAAW,kBAAmB,SAAQ,mBAAmB;IAC7D;;OAEG;IACH,QAAQ,CAAC,YAAY,EAAE,MAAM,CAAC;IAC9B;;OAEG;IACH,QAAQ,CAAC,OAAO,EAAE,MAAM,CAAC;IACzB;;;OAGG;IACH,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;IACxB;;;OAGG;IACH,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;CACzB;AAED;;GAEG;AACH,qBAAa,aAAc,SAAQ,UAAU;IAC3C,SAAS,CAAC,QAAQ,CAAC,OAAO,EAAE,MAAM,CAAC;IACnC,SAAS,CAAC,QAAQ,CAAC,YAAY,EAAE,MAAM,CAAC;IACxC,SAAS,CAAC,QAAQ,CAAC,WAAW,CAAC,EAAE,MAAM,CAAC;IACxC,SAAS,CAAC,QAAQ,CAAC,KAAK,CAAC,EAAE,MAAM,CAAC;IAClC,SAAS,CAAC,QAAQ,CAAC,KAAK,EAAE,MAAM,CAAC;gBAErB,KAAK,EAAE,eAAe,EAAE,KAAK,EAAE,kBAAkB;IAY7D,OAAO,IAAI,OAAO,EAAE;IA0BpB,SAAS,CAAC,wBAAwB,CAAC,OAAO,EAAE,MAAM;CAKnD"}