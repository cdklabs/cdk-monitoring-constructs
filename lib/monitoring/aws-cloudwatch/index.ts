export * from "./CloudWatchLogsMetricFactory";
export * from "./LogMonitoring";
