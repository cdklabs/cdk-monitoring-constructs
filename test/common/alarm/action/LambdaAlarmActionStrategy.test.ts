import { Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { Alarm, Metric } from "aws-cdk-lib/aws-cloudwatch";
import {
  determineLatestNodeRuntime,
  Function,
  InlineCode,
} from "aws-cdk-lib/aws-lambda";

import { LambdaAlarmActionStrategy } from "../../../../lib";

test("snapshot test: Lambda function", () => {
  const stack = new Stack();
  const onAlarmFunction = new Function(stack, "alarmLambda", {
    functionName: "DummyLambda",
    runtime: determineLatestNodeRuntime(stack),
    code: InlineCode.fromInline("{}"),
    handler: "Dummy::handler",
  });
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = new LambdaAlarmActionStrategy(onAlarmFunction);
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("snapshot test: Lambda alias", () => {
  const stack = new Stack();
  const onAlarmFunction = new Function(stack, "alarmLambda", {
    functionName: "DummyLambda",
    runtime: determineLatestNodeRuntime(stack),
    code: InlineCode.fromInline("{}"),
    handler: "Dummy::handler",
  });
  const alias = onAlarmFunction.addAlias("aliasName");
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = new LambdaAlarmActionStrategy(alias);
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("snapshot test: Lambda version", () => {
  const stack = new Stack();
  const onAlarmFunction = new Function(stack, "alarmLambda", {
    functionName: "DummyLambda",
    runtime: determineLatestNodeRuntime(stack),
    code: InlineCode.fromInline("{}"),
    handler: "Dummy::handler",
  });
  const version = onAlarmFunction.currentVersion;
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = new LambdaAlarmActionStrategy(version);
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
