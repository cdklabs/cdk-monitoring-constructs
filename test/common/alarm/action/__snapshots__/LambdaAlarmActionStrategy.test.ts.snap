// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: Lambda alias 1`] = `
Object {
  "Mappings": [MAPPING REMOVED],
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "alarmLambdaAliasaliasName41B27313",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "alarmLambda131DB691": Object {
      "DependsOn": Array [
        "alarmLambdaServiceRoleCDAABB9D",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "{}",
        },
        "FunctionName": "DummyLambda",
        "Handler": "Dummy::handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "alarmLambdaServiceRoleCDAABB9D",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "alarmLambdaAliasaliasName41B27313": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "alarmLambda131DB691",
        },
        "FunctionVersion": Object {
          "Fn::GetAtt": Array [
            "alarmLambdaCurrentVersionBDCE825C70892026f75c21f8b73d1d91982fbdfc",
            "Version",
          ],
        },
        "Name": "aliasName",
      },
      "Type": "AWS::Lambda::Alias",
    },
    "alarmLambdaAliasaliasNameAlarmPermission64A91652": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Ref": "alarmLambdaAliasaliasName41B27313",
        },
        "Principal": "lambda.alarms.cloudwatch.amazonaws.com",
        "SourceAccount": Object {
          "Ref": "AWS::AccountId",
        },
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "DummyAlarm234203A9",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "alarmLambdaCurrentVersionBDCE825C70892026f75c21f8b73d1d91982fbdfc": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "alarmLambda131DB691",
        },
      },
      "Type": "AWS::Lambda::Version",
    },
    "alarmLambdaServiceRoleCDAABB9D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: Lambda function 1`] = `
Object {
  "Mappings": [MAPPING REMOVED],
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::GetAtt": Array [
              "alarmLambda131DB691",
              "Arn",
            ],
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "alarmLambda131DB691": Object {
      "DependsOn": Array [
        "alarmLambdaServiceRoleCDAABB9D",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "{}",
        },
        "FunctionName": "DummyLambda",
        "Handler": "Dummy::handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "alarmLambdaServiceRoleCDAABB9D",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "alarmLambdaAlarmPermission43E41C89": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "alarmLambda131DB691",
            "Arn",
          ],
        },
        "Principal": "lambda.alarms.cloudwatch.amazonaws.com",
        "SourceAccount": Object {
          "Ref": "AWS::AccountId",
        },
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "DummyAlarm234203A9",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "alarmLambdaServiceRoleCDAABB9D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: Lambda version 1`] = `
Object {
  "Mappings": [MAPPING REMOVED],
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "alarmLambdaCurrentVersionBDCE825C70892026f75c21f8b73d1d91982fbdfc",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "alarmLambda131DB691": Object {
      "DependsOn": Array [
        "alarmLambdaServiceRoleCDAABB9D",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "{}",
        },
        "FunctionName": "DummyLambda",
        "Handler": "Dummy::handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "alarmLambdaServiceRoleCDAABB9D",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "alarmLambdaCurrentVersionAlarmPermissionFEBD056F": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Ref": "alarmLambdaCurrentVersionBDCE825C70892026f75c21f8b73d1d91982fbdfc",
        },
        "Principal": "lambda.alarms.cloudwatch.amazonaws.com",
        "SourceAccount": Object {
          "Ref": "AWS::AccountId",
        },
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "DummyAlarm234203A9",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "alarmLambdaCurrentVersionBDCE825C70892026f75c21f8b73d1d91982fbdfc": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "alarmLambda131DB691",
        },
      },
      "Type": "AWS::Lambda::Version",
    },
    "alarmLambdaServiceRoleCDAABB9D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
